projects/laji/src/app/shared/service/coordinate.service.ts

```diff
@@ -1,6 +1,38 @@
 import { Injectable } from '@angular/core';
 import * as MapUtil from 'laji-map/lib/utils';
 
+const pad = (value) => {
+  value = '' + value;
+  return value + '0000000'.slice(value.length);
+};
+
+const convertYkjToWgs = (latLng: [string, string]): [string, string] => (
+  MapUtil.convertLatLng([+latLng[0], +latLng[1]], 'EPSG:2393', 'WGS84')
+);
+
+export const convertYkjToGeoJsonFeature = (origLat: any, origLng: any, properties: {[k: string]: any} = {}) => {
+  const lat = parseInt(origLat, 10);
+  const lng = parseInt(origLng, 10);
+  const latStart = pad(lat);
+  const latEnd = pad(lat + 1);
+  const lonStart = pad(lng);
+  const lonEnd = pad(lng + 1);
+  return {
+    type: 'Feature',
+    properties,
+    geometry: {
+      type: 'Polygon',
+      coordinates: [([
+        [latStart, lonStart],
+        [latStart, lonEnd],
+        [latEnd, lonEnd],
+        [latEnd, lonStart],
+        [latStart, lonStart],
+      ] as [string, string][]).map((latLng: [string, string]) => convertYkjToWgs(latLng).reverse())],
+      coordinateVerbatim: origLat + ':' + origLng
+    }
+  };
+};
```

Comment:
The way I see it, it's confusing that there's stuff in the `shared` directory that is not declared/provided in `SharedModule`. It came up when I was trying to evaluate whether a directive that imported platformservice would require `SharedModule`. There's really no choice but to check the code of platformservice and everything that's injected into it... Imo it would be nice if the directory hierarchy and module system in combination gave a good idea of the dependencies of each component/service/w.e. In fact I think `providedIn: 'root'` might be a little bit of an antipattern too because of this...

I'm not married to the idea of moving platform service btw, but I think it's nice being able to tell that it's not a part of shared module from looking at the filepath alone.

Reply:
> Why were these moved away from the class? Having such static methods is fine and could be a reasonable option for all the functions that the coordinate service provides as it doesn't rely on any other services. but I think we should have either everything in the service class, OR a utility file that exports everything flat, but having both makes it unpredictable how the functions should he imported.

I prefer using js modules to namespace stuff over classes with static methods (the OOP way).
1. if the class has non-static methods then static methods are unnecessarily cluttering the class definition
2. class notation implies that there's shared state between the methods and static methods are basically just a workaround for pure OOP (or close to) languages (ie. not js)
2.2 to clarify my hot take further: the entire point of classes is that you can instance them. If I see a class, it's my immediate assumption that you are supposed to instance it. Static methods break this pattern.
3. static methods introduce an unnecessary level of indentation
4. functions can still be namespaced if necessary without relying on class notation (which comes with the aforementioned baggage of being associated with actual class instancing) or js modules: just define it inside a regular js object eg. **https://github.com/luomus/laji/blob/bird-atlas/projects/exhibition-screen/src/animations/math.ts**

Ie. my personal preference is to never use static methods in ts/js at all and rather leave the class definitions for things that actually use the functionality that the class keyword provides: instancing.

`CoordinateService` is particularly suspicious since it doesn't have any shared state despite all methods being non-static. There's no reason this needs to be a service. I can convert the entire class into utility functions if that's okay with you?

Reply:
Thanks for elaborating! Your points are valid and in many (and probably most) cases are the best way. I think however that there are valid cases for static methods: if you have a service that can't be static, but it has some static utility methods, putting it in the same namespace can make sense if it's needed elsewhere (makes me wonder if a private static method would ever be useful...). But even that is a matter of taste, and I'm not sure it applies for all such cases.

> CoordinateService is particularly suspicious since it doesn't have any shared state despite all methods being non-static. There's no reason this needs to be a service (or a class). I can convert the entire class into utility functions if that's okay with you?

Sounds good! It's just a collection of static utility functions so this would make a lot of sense.

Reply:
> The way I see it, it's confusing that there's stuff in the shared directory that is not declared/provided in SharedModule. It came up when I was trying to evaluate whether a directive that imported platformservice would require SharedModule. There's really no choice but to check the code of platformservice and everything that's injected into it... Imo it would be nice if the directory hierarchy and module system in combination gave a good idea of the dependencies of each component/service/w.e. In fact I think providedIn: 'root' might be a little bit of an antipattern too because of this...
> 
> I'm not married to the idea of moving platform service btw, but I think it's nice being able to tell that it's not a part of shared module from looking at the filepath alone.

Yeah, the root issue seems to be that our folder structure doesn't reflect the app structure very well... The stuff provided in `root` don't fit in either folder really. If you put them in the `shared` you might think that it's provided in the shared module. If you put it in the `shared-modules`, you might think that it's included in some module and that's also confusing. Neither place really makes sense. We should refactor the module structure... So, I don't have a strong opinion where the platform service should reside since there isn't any correct answer. Making changes that don't solve anything doesn't really help and makes following git changes more difficult, and there's already a merge conflict and the beta merge might revolt in a lot of those. In fear of merge conflicts, I'd vote on reverting the moving of the platform service.

Maybe we should have a dev-meeting about what are the current key issues that could be refactored in laji repo. Usually summer has been good time for refactoring stuff :)

We should discuss this deeper at some other point, but I'm just writing down what I'm thinking the structure could be look like:
* `modules`: current `shared-modules`
* `modules/shared` current `shared`
* `root` for stuff provided by root? You might be right that it's an antipattern to provide in the root, but bloating the shared module to include everything is maybe also not the optimal solution.

Reply:
`PlatformService` moved to `/root`

Reply:
Oops, forgot about that! Ok, `shared/service/coordinate.service` is now `root/coordinate-utils.ts` and exports bare functions. Perhaps a further consideration would if the `/root` would be renamed to `/no-module` (or perhaps `/moduleless`, or did I just play too much elden ring?) or some such, but I think '/root' is a decent enough place for the time being.

Reply:
Thanks, lookin' good! Yeah it could help if the naming somehow would implicate that it's not included in any module, but indeed it's already better so let's merge this :)